### PR TITLE
Tweak newly added spans to track formula processing.

### PR DESCRIPTION
**Pull request by samwho**
## Description

After looking at the spans in the QA environment, I added some extra tags and changed the way I'm attaching the app ID to the trace. When creating an new trace at the middleware level, it seems to become orphaned and shows up strangely in DataDog. I want to see if attaching to an existing span works better.

### CODE REVIEW

File: packages/server/src/utilities/rowProcessor/utils.ts [hunk 1]
```diff
@@ -77,7 +78,7 @@ export function processFormulas<T extends Row | Row[]>(
           rows[i] = {
             ...row,
             [column]: tracer.trace("processStringSync", {}, span => {
-              span?.addTags({ column })
+              span?.addTags({ table_id: table._id, column, statis: isStatic })
```
Comment (adrinr) [inline packages/server/src/utilities/rowProcessor/utils.ts:81]:
```suggestion
              span?.addTags({ table_id: table._id, column, static: isStatic })
```

File: packages/server/src/middleware/currentapp.ts [hunk 2]
```diff
@@ -15,109 +15,106 @@ import { UserCtx, ContextUser } from "@budibase/types"
...
+  if (!requestAppId) {
+    return next()
+  }
 
-    if (requestAppId) {
-      span?.addTags({ app_id: requestAppId })
-    }
+  if (requestAppId) {
+    const span = tracer.scope().active()
+    span?.addTags({ app_id: requestAppId })
```
Comment (adrinr) [inline packages/server/src/middleware/currentapp.ts:26]:
```suggestion
    span.addTags({ app_id: requestAppId })
```

**Review by adrinr — APPROVED** [review]
Just 2 small suggestions

**Review by shogunpurple — APPROVED** [review]

File: packages/server/src/middleware/currentapp.ts (hunk 2 above)
Comment (samwho) [reply packages/server/src/middleware/currentapp.ts:26]: My editor suggested that span may not be present, so I used the ?

How come this suggestion removes it?

Comment (samwho) [reply packages/server/src/middleware/currentapp.ts:26]: Double checked just to be sure 😅 

CleanShot 2023-12-15 at 13 14 17@2x

**Pull request merged**